pkg/endpoints/credential_test.go

```diff
@@ -0,0 +1,381 @@
+		}
+	}
+
+	namespaceSecretMap := map[string][]corev1.Secret{
+		namespaces[0]: []corev1.Secret{
+			corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "userPassSecret",
+					Annotations: map[string]string{
+						"tekton.dev/docker-0": "https://gcr.io",
+					},
+					Namespace: namespaces[0],
+				},
+				Data: map[string][]byte{
+					"username":    []byte("fakeUsername"),
+					"password":    []byte("fakePassword"),
+					"description": []byte("userPass credential"),
+				},
+				Type: corev1.SecretTypeBasicAuth,
+			},
+			corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "gitSecret",
+					Annotations: map[string]string{
+						"tekton.dev/git-0": "https://my.git.org",
+					},
+					Namespace: namespaces[0],
+				},
+				Data: map[string][]byte{
+					"username":    []byte("fakeUsername"),
+					"password":    []byte("fakePassword"),
+					"description": []byte("userPass credential"),
+				},
+				Type: corev1.SecretTypeBasicAuth,
+			},
+		},
+		namespaces[1]: []corev1.Secret{
+			corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "userPassSecret2",
+					Annotations: map[string]string{
+						"tekton.dev/docker-0": "https://gcr.io",
+					},
+					Namespace: namespaces[1],
+				},
+				Data: map[string][]byte{
+					"username":    []byte("fakeUsername"),
+					"password":    []byte("fakePassword"),
+					"description": []byte("userPass credential"),
+				},
+				Type: corev1.SecretTypeBasicAuth,
+			},
+		},
+	}
+	var allSecrets int
+
+	// Test getAllCredentials: no credentials
+	httpReq := dummyHTTPRequest("GET", fmt.Sprintf("%s/v1/namespaces/*/credentials", server.URL), nil)
+	response, err := http.DefaultClient.Do(httpReq)
+	if err != nil {
+		t.Fatalf("getAllCredentials response error: %v\n", err)
+	}
+	responseCredentialsList := []Credential{}
+	if err := json.NewDecoder(response.Body).Decode(&responseCredentialsList); err != nil {
+		t.Fatalf("Error decoding getAllPipelineRuns response: %v\n", err)
+	}
+	if len(responseCredentialsList) != 0 {
+		t.Fatal("No credentials expected")
+	}
+	// Create secrets/credentials
+	for _, namespacedSecrets := range namespaceSecretMap {
+		for _, secret := range namespacedSecrets {
+			_, err := r.K8sClient.CoreV1().Secrets(secret.Namespace).Create(&secret)
+			if err != nil {
+				t.Fatalf("Error creating secret '%s': %v\n", secret.Name, err)
+			}
+			allSecrets++
+		}
+	}
+	// Test getAllCredentials: all namespaces
+	httpReq = dummyHTTPRequest("GET", fmt.Sprintf("%s/v1/namespaces/*/credentials", server.URL), nil)
+	response, err = http.DefaultClient.Do(httpReq)
+	if err != nil {
+		t.Fatalf("getAllCredentials response error: %v\n", err)
+	}
+	responseCredentialsList = []Credential{}
+	if err := json.NewDecoder(response.Body).Decode(&responseCredentialsList); err != nil {
+		t.Fatalf("Error decoding getAllPipelineRuns response: %v\n", err)
+	}
+	if len(responseCredentialsList) != allSecrets {
```

Comment:
Instead of just checking that the number of credentials matches, please iterate over the list and check that each of the secrets has the correct content (you'll probably want a helper function for this that can be reused in this test file).